x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHexCellIdSource.java

```diff
@@ -84,6 +91,24 @@ protected int advanceValue(org.elasticsearch.common.geo.GeoPoint target, int val
         };
     }
 
+    // package private for testing
+    static LatLonBounds getGeoBounds(CellBoundary cellBoundary) {
+        final LatLonBounds bounds = new LatLonBounds();
+        org.apache.lucene.spatial3d.geom.GeoPoint A = getGeoPoint(cellBoundary.getLatLon(cellBoundary.numPoints() - 1));
+        for (int i = 0; i < cellBoundary.numPoints(); i++) {
+            final org.apache.lucene.spatial3d.geom.GeoPoint B = getGeoPoint(cellBoundary.getLatLon(i));
+            bounds.addPoint(B);
+            final Plane plane = new Plane(A, B);
+            bounds.addPlane(PlanetModel.SPHERE, new Plane(A, B), new SidedPlane(A, plane, B), new SidedPlane(B, A, plane));
```

Comment:
I guess we don't need to construct a `new Plan(A, B)`, but can use `plane` here?

Reply:
oh yes, done!